streaming/src/test/scala/org/apache/spark/streaming/scheduler/rate/PIDRateEstimatorSuite.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.streaming.scheduler.rate
+
+import scala.util.Random
+
+import org.scalatest._
+import org.scalatest.Matchers
+import org.scalatest.Inspectors._
+
+import org.apache.spark.SparkFunSuite
+
+class PIDRateEstimatorSuite extends SparkFunSuite with Matchers {
+
+  test("first bound is None") {
+    val p = new PIDRateEstimator(20, -1D, 0D, 0D)
+    p.compute(0, 10, 10, 0) should equal(None)
+  }
+
+  test("second bound is rate") {
+    val p = new PIDRateEstimator(20, -1D, 0D, 0D)
+    p.compute(0, 10, 10, 0)
+    // 1000 elements / s
+    p.compute(10, 10, 10, 0) should equal(Some(1000))
+  }
+
+  test("works even with no time between updates") {
+    val p = new PIDRateEstimator(20, -1D, 0D, 0D)
+    p.compute(0, 10, 10, 0)
```

Comment:
This is hard to understand. Basically the internal state is being mutated in these series of calls. So could you put asserts on the internal state so that I can understand what are the expected changes